### ISSUE TITLE
lsp-rust-rls-command is a lie

### ISSUE THREAD

**Issue by marcbowes** [emacs-lsp/lsp-rust#14]
The README says you can set `lsp-rust-rls-command`. However, no such variable is defined. As a result, lsp-rust failed for because my default toolchain is stable, causing `rustc ... -Z` to fail.

```
(defun lsp-rust--rls-command ()
  (let ((rls-root (getenv "RLS_ROOT"))
        ;; Look here
	(rls-path (or lsp-rust-rls-command (executable-find "rls"))))
    (if rls-path
	rls-path
      (when rls-root
	`("cargo" "+nightly" "run" "--quiet"
	  ,(concat "--manifest-path="
		   (concat
		    (file-name-as-directory (expand-file-name rls-root))
		    "Cargo.toml"))
	  "--release")))))
```

I made this small change and now things work.

**Comment by marcbowes** [comment]
I also found that I had to export RUST_SRC_PATH and RUSTC - otherwise my lsp buffer was full of errors. I landed up writing this:

```
(defun configure-rust/setup-env ()
  "RLS requires some environment variables to be setup. We use rustup to get the values."
  
  (when (executable-find "rustup")
    (require 's)
    (require 'dash)
    (setq rust-default-toolchain
          (car (s-split " " (-first
                             (lambda (line) (s-match "default" line)) 
                             (s-lines (shell-command-to-string "rustup toolchain list"))))))
    ;; tell racer to use the rustup-managed rust-src
    ;; rustup component add rust-src
    (setq rust-src-path (concat (getenv "HOME") "/.multirust/toolchains/" rust-default-toolchain "/lib/rustlib/src/rust/src"))
    (setq rust-bin-path (concat (getenv "HOME") "/.multirust/toolchains/" rust-default-toolchain "/bin"))
    (setq racer-rust-src-path rust-src-path)
    (setenv "RUST_SRC_PATH" rust-src-path)
    (setenv "RUSTC" rust-bin-path)))
```

which is used here:

```
(use-package rust-mode
  :ensure t
  ;; :mode "\\.rs\\'" ;; this is already done by rust-mode
  :config
  (add-hook 'rust-mode-hook #'lsp-rust-enable)
  (configure-rust/setup-env)
  :after lsp-mode)
```
```

**Comment by rattit0** [comment]
https://github.com/emacs-lsp/lsp-rust/blob/239e0cb8912749266a6226b68c6bc3b4362876c1/lsp-rust.el#L26
And `lsp-rust-rust-command' also doesn't exist.

**Comment by bkchr** [comment]
Could you provide a pull request that adds the variable @marcbowes?

For the RUST_SRC_PATH, yeah that variable needs to be set for racer to find the rust source code.

**Comment by fmdkdd** [comment]
Yep, the README was merged before #10 (which was blocked by an API change).  I'll look into updating #10 so we can make this right.
